File: hw/bonilla/u1/hw16Libraries/JavaScript/calculate_tax.js
import readline from "readline";
import { calculateTax } from "./IncomeTaxJavaScript.js";

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function showMenu() {
    console.log("====== INCOME TAX CALCULATOR ======");
    console.log("1. Calculate Income Tax");
    console.log("2. Exit");
    rl.question("Choose an option: ", handleMenu);
}

function handleMenu(option) {
    switch (option) {
        case "1":
            rl.question("Enter your monthly salary: $", (salary) => {
                rl.question("Enter your monthly expenses: $", (expenses) => {
                    const tax = calculateTax(parseFloat(salary), parseFloat(expenses));
                    console.log("------------------------------------");
                    console.log("Your annual income tax is: $" + tax.toFixed(2));
                    console.log("------------------------------------\n");
                    showMenu();
                });
            });
            break;

        case "2":
            console.log("Goodbye!");
            rl.close();
            break;

        default:
            console.log("Invalid option. Try again.\n");
            showMenu();
    }
}

showMenu();

File: hw/bonilla/u1/hw16Libraries/Python/calculate_tax.py
from IncomeTaxPython import calculate_tax  

def main():
    while True:
        print("====== INCOME TAX CALCULATOR ======")
        print("1. Calculate Income Tax")
        print("2. Exit")

        try:
            option = int(input("Choose an option: "))
        except ValueError:
            print("Please enter a valid number.\n")
            continue

        if option == 1:
            try:
                salary = float(input("Enter your monthly salary: $"))
                expenses = float(input("Enter your monthly expenses: $"))
                tax = calculate_tax(salary, expenses)
                print("------------------------------------")
                print(f"Your annual income tax is: ${tax:.2f}")
                print("------------------------------------\n")
            except ValueError:
                print("Invalid number. Please try again.\n")

        elif option == 2:
            print("Goodbye!")
            break
        else:
            print("Invalid option. Try again.\n")

if __name__ == "__main__":
    main()
